src/core/mount.c

```diff
@@ -1891,9 +1913,21 @@ static int mount_dispatch_io(sd_event_source *source, int fd, uint32_t revents,
                 device_found_node(m, what, 0, DEVICE_FOUND_MOUNT);
         }
 
+        m->mount_last_read = this_run;
+        m->mount_last_duration = now(CLOCK_MONOTONIC) - this_run;
+
         return 0;
 }
 
+static int mount_dispatch_timeout_io(sd_event_source *source, usec_t usec, void *userdata) {
+        Manager *m = userdata;
```

Comment:
somewhere here there should be a big comment explaining the situation and why we do this all. i.e. that this is about CPU starving out other processes when a mount storm takes place. And that this is a hack around bad kernel APIs and CPU scheduling